src/lib/server/agents/agents.ts

```diff
@@ -17,18 +20,31 @@ export const getDBAgent = async (agentId: string): Promise<DBAgent> => {
 	if (mockDbData) {
 		const foundAgent = mockDbData.agents.find((agent) => agent._id === agentId)
 		if (!foundAgent) {
-			throw new Error("Agent not found")
+			throw new HTTPError(404, `Agent ${agentId} not found`)
 		}
-		return JSON.parse(JSON.stringify(foundAgent))
+		return JSON.parse(JSON.stringify(foundAgent)) // Return a deep copy for not reference issues
 	}
 	throw new Error("Not implemented - please set MOCK_DB to true in env")
 	// Implement real DB fetch here
 }
 
-export const getDBAgents = async (): Promise<DBAgent[]> => {
+/**
+ * Handles authorization when fetching agents from DB, for faster retrieval. Probs a smarter solution somewhere...
+ * @param {AuthenticatedUser} user
+ * @returns {Promise<DBAgent[]>}
```

Comment:
FYI: These are kind of pointless. Since there's no additional documentation, TypeScript already knows all of this